src/main/java/cn/nukkit/item/ItemBucket.java

```diff
@@ -121,10 +121,13 @@ public boolean onActivate(Level level, Player player, Block block, Block target,
             Item result = Item.get(BUCKET, 0, 1);
             PlayerBucketEmptyEvent ev;
             player.getServer().getPluginManager().callEvent(ev = new PlayerBucketEmptyEvent(player, block, face, this, result));
-            ev.setCancelled(!block.canBeFlowedInto());
 
-            if (player.getLevel().getName().equals("nether") && this.getDamage() != 10) {
-                ev.setCancelled(true);
+            if(!ev.isCancelled()) {
+                ev.setCancelled(!block.canBeFlowedInto());
+
+                if (player.getLevel().getName().equals("nether") && this.getDamage() != 10) {
```

Comment:
Maybe use dimension instead of the name of the level.

Reply:
I've only added the if(!ev.isCancelled()) {